kvbc/src/v4blockchain/detail/latest_keys.cpp

```diff
@@ -120,6 +123,109 @@ void LatestKeys::handleCategoryUpdates(const std::string& block_version,
   }
 }
 
+/*
+Iterate over updates, for each key:
+1 - read its previous version by calling get with version id - 1.
+2 - if the previous version does not exist, mark the key for delete.
+3 - if the previous version exists, update the value with old value and id as version.
+*/
+void LatestKeys::revertLastBlockKeys(const concord::kvbc::categorization::Updates& updates,
```

Comment:
nit: In all these functions the write of the write_batch is done by the caller. So Actually nothing is replaced when these functions are called. Should we change the name of the function to reflect the same?
Or should we call the  native_client_->write(std::move(write_batch)); in these functions.

Reply:
as mentioned above, this operation is part of an atomic deleting the last block, so we need to use a single write batch for the whole.

Reply:
thanks @cloudnoize